### DISCUSSION TITLE
what's the meaning/function of uct_ppn_bandwidth?

### DISCUSSION THREAD

**Discussion by XCloudAI** [openucx/ucx#6978] [Q&A] [answered]
uct_ppn_bandwidth is defined belowed:
```
 873 /*
 874  * @ingroup UCT_RESOURCE
 875  * @brief Process Per Node (PPN) bandwidth specification: f(ppn) = dedicated + shared / ppn
 876  *
 877  *  This structure specifies a function which is used as basis for bandwidth
 878  * estimation of various UCT operations. This information can be used to select
 879  * the best performing combination of UCT operations.
 880  */
 881 typedef struct uct_ppn_bandwidth {
 882     double                   dedicated; /**< Dedicated bandwidth, bytes/second */
 883     double                   shared;    /**< Shared bandwidth, bytes/second */
 884 } uct_ppn_bandwidth_t;
```
What steps are included ```Process(Process Per node)```? Hwo to understand ```dedicated``` and ```shared```?

**Comment by yosefe** [comment, marked as answer]
This is estimation of endpoint's bandwidth as function of number of concurrent processes using the device. "dedicated" is bandwidth which does not depend on number of users, and "shared" is divided by number of users. Pls see https://github.com/openucx/ucx/blob/55c28859bc69c0b20520911e3af07611a1daf9a8/src/ucp/core/ucp_context.h#L443 as example

**Reply by XCloudAI** [reply]
Thanks yosefe

**Comment by changchengx** [comment]
https://github.com/openucx/ucx/blob/v1.11.0/src/uct/ib/base/ib_iface.c#L1703
